page.js

```diff
@@ -197,7 +204,15 @@ class Page {
         return evaluatedComponent
       })
 
-
+      if ('back' in request.query) {
+        progress.pop()
```

Comment:
Only pop if the last path is the currentPath (otheerewise a refresh would also pop)